server/gitlab/api.go

```diff
@@ -643,20 +643,41 @@ func (g *gitlab) GetYourProjects(ctx context.Context, user *UserInfo) ([]*intern
 	}
 	owned := true
 
-	result, resp, err := client.Projects.ListProjects(
-		&internGitlab.ListProjectsOptions{
-			Owned: &owned,
-		},
-		internGitlab.WithContext(ctx),
-	)
-	if respErr := checkResponse(resp); respErr != nil {
-		return nil, respErr
-	}
-	if err != nil {
-		return nil, err
+	var projects []*internGitlab.Project
+	if g.gitlabGroup == "" {
+		result, resp, err := client.Projects.ListProjects(
+			&internGitlab.ListProjectsOptions{
+				Owned: &owned,
+			},
+			internGitlab.WithContext(ctx),
+		)
+		if respErr := checkResponse(resp); respErr != nil {
+			return nil, respErr
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		projects = append(projects, result...)
+	} else {
+		result, resp, err := client.Groups.ListGroupProjects(
+			g.gitlabGroup,
+			&internGitlab.ListGroupProjectsOptions{
+				Owned: &owned,
+			},
+			internGitlab.WithContext(ctx),
+		)
+		if respErr := checkResponse(resp); respErr != nil {
+			return nil, respErr
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		projects = append(projects, result...)
```

Comment:
this code is repeated. can we common this out somehow?

Reply:
Discussed. This code is same in the whole plugin